src/components/Footer/Footer.js

```diff
@@ -63,23 +87,117 @@ const Footer = ({
+      await sendMessageDM(item.id, 'post', 'ANONYMOUS');
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDmAnon: false});
+    }
+  };
+
+  const dataSheet = [
+    {
+      id: 1,
+      name: loading.loadingDm ? 'Loading...' : `Message ${username}`,
+      icon: <SendDMBlack />,
+      onPress: onPressDM
+    },
+    {
+      id: 2,
+      name:
+        loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon
+          ? 'Loading...'
+          : 'Message using Incognito Mode',
+      icon: (
+        <IcDmAnon
+          color={
+            !userAllowDm || loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon
+              ? COLORS.gray
+              : 'black'
+          }
+        />
+      ),
+      onPress: userAllowDm
+        ? onPressDMAnon
+        : () => {
+            Toast.show('This user does not allow messages in Incognito Mode.', Toast.SHORT);
+          },
+      style: (!userAllowDm || loading.loadingGetAllowAnonDmStatus || loading.loadingDmAnon) && {
+        color: COLORS.gray300
+      }
+    }
+  ];
+
+  const onPressDm = async () => {
+    try {
+      refSheet.current.open();
+      setLoading({...loading, loadingGetAllowAnonDmStatus: true});
+      const data = await getAllowAnonDmStatus(item.id);
+      setUserAllowDm(data?.user.allow_anon_dm);
+    } catch (e) {
+      console.log(e);
+    } finally {
+      setLoading({...loading, loadingGetAllowAnonDmStatus: false});
+    }
+  };
+
   return (
     <BlurredLayer toastOnly={true} isVisible={item?.isBlurredPost}>
       <View style={[styles.rowSpaceBetween, styles.container]}>
         <View style={styles.leftGroupContainer}>
-          <TouchableOpacity
-            testID="shareBtn"
-            disabled={item?.isBlurredPost}
-            style={styles.btn}
-            onPress={onPressShare}>
-            <View style={styles.btnShare}>
-              <MemoIc_share
-                color={item?.isBlurredPost ? COLORS.gray : undefined}
-                height={20}
-                width={21}
-              />
-            </View>
-          </TouchableOpacity>
+          {isShowDM && !isSelf ? (
+            <TouchableOpacity
+              testID="sendDM"
+              style={[
+                {
+                  width: 100
+                }
+              ]}
+              onPress={onPressDm}>
+              <View style={[styles.btnShare, styles.buttonShareContainer]}>
+                <MemoIc_senddm height={20} width={21} />
+                <Text style={[styles.text, styles.textDM]}>DM</Text>
+              </View>
+            </TouchableOpacity>
+          ) : (
+            <TouchableOpacity testID="shareBtn" style={styles.btn} onPress={onPressShare}>
+              <View style={styles.btnShare}>
+                <MemoIc_share height={20} width={21} />
+              </View>
+            </TouchableOpacity>
+          )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The render function has been updated to include the new DM button and BottomSheetMenu component. The DM button is only shown if `isShowDM` is true and the user is not viewing their own post (`!isSelf`). The BottomSheetMenu is used to provide options for sending direct or anonymous messages. These changes seem appropriate and follow React best practices.


<!-- This is an auto-generated comment by OSS CodeRabbit -->